File: common/src/main/java/com/dreamdisplays/ytdlp/VideoTitleCache.java
package com.dreamdisplays.ytdlp;

import org.jspecify.annotations.NullMarked;
import org.jspecify.annotations.Nullable;

import java.util.concurrent.ConcurrentHashMap;

@NullMarked
public final class VideoTitleCache {

    private static final ConcurrentHashMap<String, String> TITLES = new ConcurrentHashMap<>();

    public static void put(String videoId, String title) {
        if (videoId.isEmpty() || title.isEmpty()) return;
        TITLES.put(videoId, title);
    }

    public static @Nullable String get(String videoId) {
        return TITLES.get(videoId);
    }
}

File: common/src/main/kotlin/com/dreamdisplays/ytdlp/VideoTitleCache.kt
package com.dreamdisplays.ytdlp

import java.util.concurrent.ConcurrentHashMap

/** Cache for video titles, to avoid repeated calls to `yt-dlp`. */
object VideoTitleCache {

    private val TITLES = ConcurrentHashMap<String, String>()


    fun put(videoId: String, title: String) {
        if (videoId.isEmpty() || title.isEmpty()) return
        TITLES[videoId] = title
    }


    fun get(videoId: String): String? = TITLES[videoId]
}
